lib/rubocop/cop/sorbet/signatures/signature_cop.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # Abstract cop specific to Sorbet signatures
+      #
+      # You can subclass it to use the `on_signature` trigger and the `signature?` node matcher.
+      class SignatureCop < RuboCop::Cop::Cop
+        @registry = Cop.registry # So we can properly subclass this cop
+
+        def_node_matcher(:signature?, <<~PATTERN)
+          (block (send nil? :sig) (args) ...)
+        PATTERN
+
+        def on_block(node)
+          on_signature(node) if signature?(node)
+        end
+
+        def on_signature(_)
+          # To be defined in subclasses
```

Comment:
Should we raise a `NotImplementedError` to fail fast in case someone tries to use the cop on its own?

Reply:
You could be subclassing this only to get the `signature?` node matcher without wanting to trigger something on each `sig` found.

For example [this cop](https://github.com/Shopify/rubocop-sorbet/pull/13/files#diff-f9ca2a82c8f0dbba33cb5736d1199804R34-R45), while it handles `sig` and use the matcher, don't need the `on_signature` trigger. Adding a raise `NotImplementedError` would make it throw errors when everything is perfectly fine.